packages/block-editor/src/components/inner-blocks/index.js

```diff
@@ -169,8 +169,11 @@ const ForwardedInnerBlocks = forwardRef( ( props, ref ) => {
  * @see https://github.com/WordPress/gutenberg/blob/HEAD/packages/block-editor/src/components/inner-blocks/README.md
  */
 export function useInnerBlocksProps( props = {}, options = {} ) {
-	const { __unstableDisableLayoutClassNames, __unstableDisableDropZone } =
-		options;
+	const {
+		__unstableDisableLayoutClassNames,
+		__unstableDisableDropZone,
+		__unstableDropZoneElement,
```

Comment:
If we want to make this private, prefixing is not the way anymore. I think we're now avoiding any new `__unstable` or `__experimental` prefix.

Reply:
I just read your message above, you should use the lock/unlock API to mark this private. These days prefixing or not doesn't impact our policies for new APIs.

Reply:
Thanks for the suggestion and extra context @youknowriad! I've had a go at doing this over in https://github.com/WordPress/gutenberg/pull/56308, would love your feedback to see if that's what you had in mind. Due to how `useInnerBlocksProps` is written and used, the solution wound up being a little indirect.

A potential alternative would be for us to stabilise `dropZoneElement` for `useInnerBlocksProps` and `BlockList` if we're comfortable with it as an API. We could then still look into further follow-ups that abstract it away a little further for the post and site editors (or folks building custom block editors).

Reply:
If we're making this private with the intention of removing it later and making the drop zone part of BlockCanvas as suggested above, my main question as to whether we should go one way or the other is if it would be useful at all to be able to set a drop zone in inner blocks. #26049 has been open for 3 years with no comments and few links to it, and the issue itself only links to one potential use case, so there doesn't seem to be a huge demand for such an API.

Reply:
> https://github.com/WordPress/gutenberg/issues/26049 has been open for 3 years with no comments and few links to it, and the issue itself only links to one potential use case, so there doesn't seem to be a huge demand for such an API.

I suppose the presence of that issue nudges me slightly toward "let's go for private for now" as it means we can then continue to explore the idea of allowing it for (likely 3rd party) container blocks in follow-up PRs (by digging into the potential use case). If we find it isn't useful, we can abandon it and go with making it part of `BlockCanvas`, and if it is useful, we can make the API public (and also potentially still explore making it an internal implementation detail of `BlockCanvas`). In both cases, it sounds like the private approach in #56308 could make for a decent interim place for now? I agree it doesn't sound like there's much demand for it for core container blocks to set their own element 🤔

I don't feel strongly about which way we go, though, my main interest is getting us to an interim state we feel comfortable with for now 🙂

Reply:
> https://github.com/WordPress/gutenberg/issues/26049 has been open for 3 years with no comments and few links to it, and the issue itself only links to one potential use case, so there doesn't seem to be a huge demand for such an API.

A lot of folks complain generally about the drag and drop experience being clunky without really providing specifics or knowing the underlying reason. I don't think I did a good job of writing that issue, I should really have led with the problem rather than a solution, and I think that has meant not many comments.

> and the issue itself only links to one potential use case

There are a few blocks it happens with - Cover, Media & Text and Group. It happens any time there's padding around the inner blocks element.

Reply:
> It happens any time there's padding around the inner blocks element.

Ah, excellent. Thanks for confirming, Dan! That gives us something concrete to investigate here 👍

Reply:
I can reproduce with the Cover block. I've added a screengrab to https://github.com/WordPress/gutenberg/issues/26049#issuecomment-1818202943 to demonstrate the problem. I'm happy to pick up that task next as it's very related to this PR.

Reply:
Another update: given that solutions to #26049 will likely involve the `dropZoneElement` prop on `useInnerBlocksProps`, I've opened up another PR to stabilise the `dropZoneElement` prop, over in #56313, as an alternative to making it private. We can likely go either way, but stabilising is the simpler code change.